lua/neo-tree/sources/filesystem/lib/shared_clipboard.lua

```diff
@@ -0,0 +1,92 @@
+local fs_watch = require("neo-tree.sources.filesystem.lib.fs_watch")
+local manager = require("neo-tree.sources.manager")
+local events = require("neo-tree.events")
+local renderer = require("neo-tree.ui.renderer")
+
+local M = {}
+
+local clipboard_state_dir_path = vim.fn.stdpath("state") .. "/neo-tree/"
+local clipboard_file_path = clipboard_state_dir_path .. "filesystem-clipboard.json"
+local clipboard_file_last_mtime = nil
+
+
+M.save_clipboard = function(clipboard)
+  local file = io.open(clipboard_file_path, "w+")
+  -- We want to erase data in the file if clipboard is nil instead writing null
+  if not clipboard or not file then
+    return
+  end
+
+  file:write(vim.json.encode(clipboard))
+  file:flush()
+  clipboard_file_last_mtime = vim.uv.fs_stat(clipboard_file_path).mtime
+end
+
+M._load_clipboard = function()
+  local file = io.open(clipboard_file_path, "r")
+  if not file then
+    return nil
+  end
+  local content = file:read("*a")
+  local is_success, clipboard = pcall(vim.json.decode, content)
+  if not is_success then
+    return nil
+  end
+  return clipboard
+end
+
+M._update_cilpboard = function()
+  if not vim.fn.filereadable(clipboard_file_path) then
+    return
+  end
+  local cur_mtime = vim.uv.fs_stat(clipboard_file_path).mtime
+  if
+    not clipboard_file_last_mtime
+    -- We need exactly >= because it allows us to synchronize the clipboard
+    -- even with trees openned in another window (in the current Neovim instance)
+    or M._is_left_mtime_greater_or_equal(cur_mtime, clipboard_file_last_mtime)
```

Comment:
It seems like a bad idea to read from the file we just wrote as a means of syncing within an instance. 

I would just extract out the update logic so it can also be called from `save_clipboard` to handle the internal syncing, then make this a greater than (but not equal to) check.

OR: Another way to handle this which may be cleaner is to have each instance write it's own file, maybe use the process ID or something to identify it. Then you can ignore your own file from the event and just handle any other events that are thrown regardless of the modified time.